.kokoro/deps-finder.sh

```diff
@@ -0,0 +1,143 @@
+
+# Pass in dependency to script.
+# Dependency should be of the form artifactId:version
+
+function checkLibrary() {
+  # Should be of the format artifactId:version
+  clientLibrary=$1
+
+  # If no dependency was passed straight to the script, check our most recent commit for a dep version update.
+  if [[ -z ${clientLibrary} ]]; then
+    clientLibrary=$(getLibraryUpdate)
+    # We did not successfully find a valid commit with a dependency. Return success - nothing to check here!
+    if ! [[ $? -eq 0 ]]; then
+      echo "Commit message does not have correct formatting. Assuming no error."
+      return 0
+    fi
+  else
+    # Check if dependency contains 'google-cloud-'. If it does not, it's already valid!
+    if [[ -z $(echo ${clientLibrary} | grep "google-cloud-") ]]; then
+      echo "${clientLibrary} does not require shared-dependencies criteria. Returning success."
+      return 0
+    fi
+  fi
+
+  # Should just be the desired version for shared-deps, e.g. "0.8.3"
+  latestSharedDeps=$2
+
+  if [[ -z ${latestSharedDeps} ]]; then
+    latestSharedDeps=$(getLatestSharedDepsVersion)
+    if [[ -z ${latestSharedDeps} ]]; then
+      echo "Unable to find correct version of shared-dependencies."
+      return 1
+    fi
+  fi
+
+  # Grab last part of dependency artifactId:version input.
+  # Remove any alphabetical characters at the end (e.g. -alpha, -beta, etc)
+  depVersion=${clientLibrary/*:/}
+  depVersion=${depVersion/-*/}
+
+  pomLocation="/pom.xml"
+
+  # The naming of these libraries does not follow the rules. As such, we've added in special cases for them.
+  # depArtifactId => the ID inside of the Github link for the dependency.
+  case ${clientLibrary} in
+  *"google-cloud-bigtable-bom"*)
+    depArtifactId="java-bigtable"
+    pomLocation="/google-cloud-bigtable-deps-bom/pom.xml"
+    ;;
+  *"google-cloud-core-bom"*)
+    # This is an exceptional case which does not require shared-dependencies! Return true automatically.
+    echo "Found exception google-cloud-core-bom!"
+    return 0
+    ;;
+  *"google-cloud-build-bom"*)
+    depArtifactId="java-cloudbuild"
+    ;;
+  *"google-cloud-monitoring-dashboard-bom"*)
+    depArtifactId="java-monitoring-dashboards"
+    ;;
+  *"google-cloud-nio"*)
+    depArtifactId="java-storage-nio"
+    ;;
+  *)
+    # General rules for obtaining the Github repo name are:
+    # Replace google-cloud with java, remove the version number, and remove -bom from the end (if there)
+    depArtifactId=${clientLibrary/*google-cloud-/java-}
+    depArtifactId=${depArtifactId/:*/}
+    depArtifactId=${depArtifactId/-bom*/}
+    ;;
+  esac
+  # Grab raw POM file from Github
+  rawFileStart="https://raw.githubusercontent.com/googleapis/"
+  gitPom=${rawFileStart}${depArtifactId}"/v"${depVersion}${pomLocation}
+
+  # If $3 is nonempty, we're running this script directly.
+  if [[ ! -z $3 ]]; then
+    echo "POM File Link Generated: ${gitPom}"
+  fi
+
+  content=$(wget ${gitPom} -q -O -)
```

Comment:
suggestion: curl is more commonly available by default on linux installs.

Reply:
After testing grabbing the POM from Git using the release-versions script, it seemed that curl was more likely to fail (at least on my personal system), and failed roughly 1/70 times and made the script take significantly longer to run. For those reasons, I switched the script back to using wget.